cmd/snap-update-ns/utils.go

```diff
@@ -286,6 +291,63 @@ func secureMkfileAll(name string, perm os.FileMode, uid, gid int) error {
 	return err
 }
 
+func planWritableMimic(dir string) ([]*Change, error) {
+	// We need a place for "safe keeping" of what is present in the original
+	// directory as we are about to attach a tmpfs there, which will hide
+	// everything inside.
+	logger.Debugf("create-writable-mimic %q", dir)
+	safeKeepingDir := filepath.Join("/tmp/.snap/", dir)
+
+	var changes []*Change
+
+	// Bind mount the original directory elsewhere for safe-keeping.
+	changes = append(changes, &Change{
+		Action: Mount, Entry: mount.Entry{
+			// XXX: should we rbind here?
+			Name: dir, Dir: safeKeepingDir, Options: []string{"bind"}},
+	})
+	// Mount tmpfs over the original directory, hiding its contents.
+	changes = append(changes, &Change{
+		Action: Mount, Entry: mount.Entry{Name: "none", Dir: dir, Type: "tmpfs"},
+	})
```

Comment:
```
22:55 < zyga-ubuntu> jdstrand: so I maybe agree if your proposal is undo-safe
22:56 < zyga-ubuntu> jdstrand: note that we may be asked to completely undo any of this
22:56 < zyga-ubuntu> jdstrand: and that the typical situation, nothing is hidden under the tmpfs mount
22:56 < zyga-ubuntu> jdstrand: (apart from a fragment of a squashfs mount)
22:56 < zyga-ubuntu> jdstrand: the simple approach I took guarantees (I hope, prove me wrong please) that I can always undo
22:56 < zyga-ubuntu> jdstrand: and create another layout
22:58 < jdstrand> zyga-ubuntu: that might be a case for needing the contents of the mounted over directory
23:00 < zyga-ubuntu> jdstrand: I also toyed with one more idea
23:00 < zyga-ubuntu> jdstrand: iff it's a good assumption that we can look at hostfs, we can fish the snap mounts there
23:00 < zyga-ubuntu> jdstrand: but it's not generic as the same code should be useful for poking holes in various places that are not just /snap 
23:01 < zyga-ubuntu> jdstrand: and while it's possible to find the same place in hostfs it may have been changed by the hard-coded mount commands in snap-confine 
```